archive/2021-04-08-New-End-ESM-Clipper-LINKA.t.sol

```diff
@@ -0,0 +1,2087 @@
+        (, rate, spot,,) = vat.ilks("LINK-A");
+        vat.frob("LINK-A", address(this), address(this), address(this), int256(ilkAmt), int256(mul(ilkAmt, spot) / rate));
+
+        hevm.warp(block.timestamp + 1);
+        jug.drip("ETH-A");
+        jug.drip("BAT-A");
+        jug.drip("LINK-A");
+
+        uint256 auctionIdETHA = flipETHA.kicks() + 1;
+        uint256 auctionIdBATA = flipBATA.kicks() + 1;
+        uint256 auctionIdLINKA = clipLINKA.kicks() + 1;
+
+        cat.bite("ETH-A", address(this));
+        cat.bite("BAT-A", address(this));
+        dog.bark("LINK-A", address(this), address(this));
+
+        assertEq(flipETHA.kicks(), auctionIdETHA);
+        assertEq(flipBATA.kicks(), auctionIdBATA);
+        assertEq(clipLINKA.kicks(), auctionIdLINKA);
+
+        hevm.store(
+            address(end),
+            keccak256(abi.encode(address(this), uint256(0))),
+            bytes32(uint256(1))
+        );
+        assertEq(end.wards(address(this)), 1);
+
+        end.cage();
+        end.cage("ETH-A");
+        end.cage("BAT-A");
+        end.cage("LINK-A");
+
+        (,, address usr,,,,,) = flipETHA.bids(auctionIdETHA);
+        assertTrue(usr != address(0));
+        (,, usr,,,,,) = flipBATA.bids(auctionIdBATA);
+        assertTrue(usr != address(0));
+        (,,, usr,,) = clipLINKA.sales(auctionIdLINKA);
+        assertTrue(usr != address(0));
+
+        end.skip("ETH-A", auctionIdETHA);
+        end.skip("BAT-A", auctionIdBATA);
+        end.snip("LINK-A", auctionIdLINKA);
+
+        (,, usr,,,,,) = flipETHA.bids(auctionIdETHA);
+        assertTrue(usr == address(0));
+        (,, usr,,,,,) = flipBATA.bids(auctionIdBATA);
+        assertTrue(usr == address(0));
+        (,,, usr,,) = clipLINKA.sales(auctionIdLINKA);
+        assertTrue(usr == address(0));
+
+        end.skim("ETH-A", address(this));
+        end.skim("BAT-A", address(this));
+        end.skim("LINK-A", address(this));
+
+        end.free("ETH-A");
+        end.free("BAT-A");
+        end.free("LINK-A");
+
+        hevm.warp(block.timestamp + end.wait());
+
+        vow.heal(min(vat.dai(address(vow)), sub(sub(vat.sin(address(vow)), vow.Sin()), vow.Ash())));
+
+        end.thaw();
+
+        end.flow("ETH-A");
+        end.flow("BAT-A");
+        end.flow("LINK-A");
+
+        vat.hope(address(end));
+
+        uint256 daiToRedeem = vat.dai(address(this)) / RAY;
+        assertTrue(daiToRedeem > 0);
+
+        end.pack(daiToRedeem);
+
+        end.cash("ETH-A", daiToRedeem);
+        end.cash("BAT-A", daiToRedeem);
+        end.cash("LINK-A", daiToRedeem);
+    }
+
+    function testSpellIsCast_new_ClipperMom() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        assertEq(clipMom.owner(), address(pauseProxy));
+        assertEq(clipMom.authority(), address(chief));
```

Comment:
check constructor param of clipperMom: 

    assertEq(clipMom.spotter(), address(spotter));